tools/wpt/tests/test_wpt.py

```diff
@@ -91,6 +91,24 @@ def test_files_changed(capsys):
     assert err == ""
 
 
+def test_files_changed_ignore():
+    from tools.wpt.testfiles import exclude_ignored
+    files = ["resources/testharness.js", "resources/webidl2/index.js", "test/test.js"]
+    changed, ignored = exclude_ignored(files, ignore_rules=["resources/testharness*"])
+    assert changed == [os.path.join(wpt.wpt_root, item) for item in
+                       ["resources/webidl2/index.js", "test/test.js"]]
+    assert ignored == [os.path.join(wpt.wpt_root, item) for item in
+                       ["resources/testharness.js"]]
+
+
+def test_files_changed_ignore_rules():
+    from tools.wpt.testfiles import compile_ignore_rule
+    assert compile_ignore_rule("foo*bar*/baz").pattern == "^foo\*bar[^/]*/baz$"
```

Comment:
Is this really right? Why is `\*` part of the pattern here or anywhere, and not `.*`?

Reply:
It's just doing prefix matching on path parts. It could do something different, but this is enough to satisfy the current use cases.

Reply:
Oh, the comment "* before a separator or the end matching" explains this.